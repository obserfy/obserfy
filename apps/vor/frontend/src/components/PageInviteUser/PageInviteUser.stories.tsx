import { FC } from "react"
import PageInviteUser from "./PageInviteUser"

export default {
  title: "Core/PageInviteUser",
  component: PageInviteUser,
  parameters: {
    componentSubtitle: "Just a simple PageInviteUser",
  },
}

export const Basic: FC = () => <PageInviteUser />
