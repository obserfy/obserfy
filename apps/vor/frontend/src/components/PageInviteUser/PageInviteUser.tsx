import { FC } from "react"
import { Box, Button, Flex } from "theme-ui"
import { nanoid } from "nanoid"
import { useImmer } from "use-immer"
import { t, Trans } from "@lingui/macro"
import BackNavigation from "../BackNavigation/BackNavigation"
import { ADMIN_URL } from "../../routes"
import Typography from "../Typography/Typography"
import { ReactComponent as CloseIcon } from "../../icons/close.svg"
import { ReactComponent as PlusIcon } from "../../icons/plus.svg"

import Icon from "../Icon/Icon"
import Input from "../Input/Input"
import { usePostUserInvite } from "../../hooks/api/usePostUserInvite"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import { useGetSchool } from "../../hooks/api/schools/useGetSchool"

const PageInviteUser: FC = () => {
  const schoolDetail = useGetSchool()
  const [emails, setEmails] = useImmer([{ id: nanoid(), email: "" }])
  const { mutateAsync, status } = usePostUserInvite()

  const removeItem = (id: string): void => {
    setEmails((draft) =>
      draft.filter((email) => {
        return email.id !== id
      })
    )
  }

  const editItem = (id: string, email: string): void => {
    const index = emails.findIndex((el) => el.id === id)
    if (email) {
      setEmails((draft) => {
        draft[index].email = email
      })
    }
  }

  const sendInvitation = async (): Promise<void> => {
    try {
      await mutateAsync({ email: emails.map((el) => el.email) })
      setEmails(() => [{ id: nanoid(), email: "" }])
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  function shareLink(): void {
    if (window.navigator?.share) {
      window.navigator?.share({
        title: `Join ${schoolDetail.data?.name} in Obserfy`,
        text: "Check out obserfy. Manage your student data.",
        url: schoolDetail.data?.inviteLink,
      })
    }
  }

  return (
    <Box sx={{ maxWidth: "maxWidth.sm" }} mx="auto">
      <BackNavigation to={ADMIN_URL} text={t`Settings`} />
      <Typography.H6 m={3} mb={4} sx={{ lineHeight: 1 }}>
        <Trans>Invite using emails</Trans>
      </Typography.H6>
      {emails?.map(({ id, email }) => (
        <EmailInput
          key={id}
          editItem={editItem}
          removeFromEmailList={removeItem}
          email={email}
          id={id}
        />
      ))}
      <Box mx={3}>
        <Button
          mb={3}
          variant="outline"
          sx={{ width: "100%" }}
          color="onBackground"
          onClick={() =>
            setEmails((draft) => {
              draft.push({ id: nanoid(), email: "" })
            })
          }
        >
          <Icon as={PlusIcon} mr={2} fill="textPrimary" />
          <Trans>Add more email</Trans>
        </Button>
        <Button sx={{ width: "100%" }} onClick={sendInvitation}>
          <Trans>Send invites</Trans>
        </Button>
      </Box>
      {status === "success" && (
        <Typography.Body
          m={3}
          sx={{
            textAlign: "center",
            color: "primary",
            fontWeight: "bold",
          }}
        >
          <Trans>Emails sent!</Trans>
        </Typography.Body>
      )}
      <Typography.Body my={3} sx={{ textAlign: "center" }}>
        <Trans>Or</Trans>
      </Typography.Body>
      <Box
        m={3}
        p={3}
        sx={{
          backgroundColor: "surface",
          borderRadius: "default",
          alignItems: "center",
          borderStyle: "solid",
          borderColor: "border",
          borderWidth: 1,
        }}
      >
        <Typography.Body color="textMediumEmphasis" sx={{ fontSize: 1 }}>
          <Trans>Invite using link</Trans>
        </Typography.Body>
        <Typography.Body mb={3}>
          Share this link to your co-workers. Once they register using it,
          they&apos;ll have access to {schoolDetail.data?.name}.
        </Typography.Body>
        {schoolDetail.status === "loading" && (
          <LoadingPlaceholder sx={{ width: "100%", height: 60 }} />
        )}
        <Typography.Body sx={{ width: "100%" }} onClick={shareLink}>
          {schoolDetail.data?.inviteLink}
        </Typography.Body>
      </Box>
    </Box>
  )
}

const EmailInput: FC<{
  id: string
  email?: string
  editItem: Function
  removeFromEmailList: Function
}> = ({ id, editItem, removeFromEmailList }) => {
  return (
    <Flex px={3} mb={2}>
      <Button variant="outline" onClick={() => removeFromEmailList(id)}>
        <Icon as={CloseIcon} fill="danger" />
      </Button>
      <Input
        sx={{ width: "100%" }}
        name="email"
        ml={2}
        placeholder="Email address"
        onChange={(e) => editItem(id, e.target.value)}
      />
    </Flex>
  )
}

export default PageInviteUser
