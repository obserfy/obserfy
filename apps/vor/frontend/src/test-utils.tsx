/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
