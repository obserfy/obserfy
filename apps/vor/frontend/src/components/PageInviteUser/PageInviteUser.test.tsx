import { render } from "../../test-utils"
import { Basic } from "./PageInviteUser.stories"

describe("PageInviteUser", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
