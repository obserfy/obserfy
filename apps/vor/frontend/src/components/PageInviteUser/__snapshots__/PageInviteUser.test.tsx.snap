// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageInviteUser should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin: 16px;
  margin-right: 4px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 20px;
  margin: 16px;
  margin-bottom: 32px;
  line-height: 1;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  margin-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
}

.emotion-7:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-7:hover,
.emotion-7:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-7:hover:disabled,
.emotion-7:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-danger);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-left: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-11:autofill,
.emotion-11:autofill:hover,
.emotion-11:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-11:autofill:first-line,
.emotion-11:autofill:hover:first-line,
.emotion-11:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-11:-webkit-autofill,
.emotion-11:-webkit-autofill:hover,
.emotion-11:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-11:-webkit-autofill:first-line,
.emotion-11:-webkit-autofill:hover:first-line,
.emotion-11:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-11:hover,
.emotion-11:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-11::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-11::-moz-placeholder {
  font-style: italic;
}

.emotion-11:-ms-input-placeholder {
  font-style: italic;
}

.emotion-11::placeholder {
  font-style: italic;
}

.emotion-11:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: 16px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-bottom: 16px;
  color: var(--theme-ui-colors-onBackground);
  width: 100%;
}

.emotion-13:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-13:hover,
.emotion-13:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-13:hover:disabled,
.emotion-13:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 8px;
  fill: var(--theme-ui-colors-textPrimary);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  width: 100%;
}

.emotion-15:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-15:hover,
.emotion-15:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-15:hover:disabled,
.emotion-15:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-top: 16px;
  margin-bottom: 16px;
  text-align: center;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
  padding: 16px;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-style: solid;
  border-color: var(--theme-ui-colors-border);
  border-width: 1px;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-size: 14px;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 16px;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 60px;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        href="/dashboard/admin"
        style="display: block;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <p
            class="emotion-4"
          >
            Settings
          </p>
        </div>
      </a>
      <h6
        class="emotion-5"
      >
        Invite using emails
      </h6>
      <div
        class="emotion-6"
      >
        <button
          class="emotion-7"
        >
          <div
            class="emotion-8"
          />
        </button>
        <label
          class="emotion-9"
        >
          <div
            class="emotion-10"
          >
            <input
              class="emotion-11"
              name="email"
              placeholder="Email address"
              value=""
            />
          </div>
        </label>
      </div>
      <div
        class="emotion-12"
      >
        <button
          class="emotion-13"
        >
          <div
            class="emotion-14"
          />
          Add more email
        </button>
        <button
          class="emotion-15"
        >
          Send invites
        </button>
      </div>
      <p
        class="emotion-16"
      >
        Or
      </p>
      <div
        class="emotion-17"
      >
        <p
          class="emotion-18"
        >
          Invite using link
        </p>
        <p
          class="emotion-19"
        >
          Share this link to your co-workers. Once they register using it, they'll have access to 
          .
        </p>
        <div
          class="emotion-20"
        />
        <p
          class="emotion-21"
        />
      </div>
    </div>
  </div>
</div>
`;
